src/fusets/peakvalley.py

```diff
@@ -0,0 +1,168 @@
+from datetime import datetime
+from typing import Iterable, Sequence, Tuple
+
+import numpy as np
+import xarray
+from scipy.signal import find_peaks
+from xarray import DataArray
+
+from fusets._xarray_utils import _extract_dates, _time_dimension
+
+
+def peakvalley(
+    array: DataArray,
+    drop_thr: float = 0.15,
+    rec_r: float = 1.0,
+    slope_thr: float = -0.007,
+) -> DataArray:
+
+    dates = np.array(_extract_dates(array))
+    time_dimension = _time_dimension(array, None)
+
+    def callback(timeseries):
+        out, _ = peakvalley_f(dates, timeseries, drop_thr, rec_r, slope_thr)
+        return out
+
+    result = xarray.apply_ufunc(
+        callback,
+        array,
+        input_core_dims=[[time_dimension]],
+        output_core_dims=[[time_dimension]],
+        vectorize=True,
+    )
+
+    result = result.rename("peak_valley_mask")
+    return result
+
+
+def peakvalley_f(
+    x: Sequence[datetime],
+    y: np.ndarray,
+    drop_thr: float = 0.15,
+    rec_r: float = 1.0,
+    slope_thr: float = -0.007,
+) -> np.ndarray:
+    """
+    Algorithm for finding peak-valley patterns in the provided array.
+
+    @param x: array of timestamps
```

Comment:
Note that our documentation generation tool expects 'Google Style' python docs, I recently adjusted some other functions as well.

Reply:
I changed it according to https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html